app/controllers/search_controller.rb

```diff
@@ -140,6 +154,19 @@ def site_overloaded?
       (queue_time > GlobalSetting.disable_search_queue_threshold)
   end
 
+  def rate_limit_search
+    begin
+      if current_user.present?
+        RateLimiter.new(current_user, "search-min", SiteSetting.rate_limit_search_user, 1.minute).performed!
+      else
+        RateLimiter.new(nil, "search-min-#{request.remote_ip}", SiteSetting.rate_limit_search_anon, 1.minute).performed!
+      end
+    rescue RateLimiter::LimitExceeded => e
```

Comment:
Hmm this is a little odd, we should just throw the error and let the `rescue_from` handler in `ApplicationController` handle it.